### PR TITLE
Compclasses cleanup

### LINKED ISSUE #1170: get_timing.py specifies component classes by hand
Are lines like this:
`for k in ['CPL', 'GLC', 'WAV', 'LND', 'ROF', 'ICE', 'ATM', 'OCN']:`
appropriate in get_timing.py? They prevent get_timing from automatically picking up new component types (e.g., ESP).

### PR DESCRIPTION
**Pull request by jedwards4b**
Get rid of hard coded comp_class settings and use get_values instead.
Punting on NCR test - it's broken, was broken before this PR will open a ticket. 

Test suite: scripts_regression_tests.py, hand tests of PET and PEM tests.         PEM.f19_g16.F1850.yellowstone_intel
      PET.f19_g16.F1850.yellowstone_intel
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #1170 

User interface changes?: 

Code review: gold2718

### CODE REVIEW

File: scripts/lib/CIME/test_scheduler.py [hunk 1]
```diff
@@ -218,13 +218,15 @@ def __init__(self, test_names, test_data=None,
                         else:
                             self._update_test_status(test, phase, TEST_PEND_STATUS)
                             self._update_test_status(test, phase, status)
+                logger.info("Using existing test directory %s"%(self._get_test_dir(test)))
```
Comment (jedwards4b) [inline scripts/lib/CIME/test_scheduler.py]: Unrelated change to improve output from create_test

**Comment by gold2718** [comment]
Why are the PET and PEM tests not tested in scripts_regression_tests.py (I know, not really part of this PR review but I'm curious)?

File: scripts/lib/CIME/test_scheduler.py (hunk 1 above)
Comment (gold2718) [inline scripts/lib/CIME/test_scheduler.py]: Since this log output is inside the self._tests loop, wouldn't you end up with identical 'Using existing test directory' log entries if there is more than one test? I suggest either adding the test name to the log output (see below) or move the log outside the for loop.
`"Using existing test directory %s for "%(self._get_test_dir(test), test)`

File: scripts/lib/CIME/test_scheduler.py [hunk 2]
```diff
@@ -218,13 +218,15 @@ def __init__(self, test_names, test_data=None,
...
         else:
             # None of the test directories should already exist.
             for test in self._tests:
                 expect(not os.path.exists(self._get_test_dir(test)),
                        "Cannot create new case in directory '%s', it already exists."
                        " Pick a different test-id" % self._get_test_dir(test))
-        logger.info("Created test in directory %s"%self._get_test_dir(test))
+                logger.info("Creating test directory %s"%(self._get_test_dir(test)))
+
```
Comment (gold2718) [inline scripts/lib/CIME/test_scheduler.py]: As above, since you moved the statement inside the loop, please add the test name to the log.

Comment (jedwards4b) [inline scripts/lib/CIME/test_scheduler.py]: Try before you comment - the output from _get_test_dir includes the test name.

File: scripts/lib/CIME/test_scheduler.py (hunk 1 above)
Comment (jedwards4b) [inline scripts/lib/CIME/test_scheduler.py]: The output from _get_test_dir includes the test name.

**Comment by jedwards4b** [comment]
Regarding the PET and PEM tests - they aren't in cime standalone tests because none of the cime standalone code includes threading and so they wouldn't do anything.

File: scripts/lib/CIME/test_scheduler.py (hunk 2 above)
Comment (gold2718) [inline scripts/lib/CIME/test_scheduler.py]: Oy, yes the test name is buried in the directory name, however, that's not what the user typed. It was just a suggestion to aid users who may not be used to parsing the test directory names.

Comment (jedwards4b) [inline scripts/lib/CIME/test_scheduler.py]: This change was in response to a user request that just seeing the test name here was not enough and they wanted the full path to the test.   I think that the test name is output elsewhere.

**Comment by jedwards4b** [comment]
I have updated the documentation to reflect the PET and PEM tests that I ran.
